pkg/daemon/writer.go

```diff
@@ -142,32 +142,40 @@ func (nw *NodeWriter) SetSSHAccessed(client corev1.NodeInterface, node string) e
 // number of times.
 // f will be called each time since the node object will likely have changed if
 // a retry is necessary.
-func updateNodeRetry(client corev1.NodeInterface, node string, f func(*v1.Node)) error {
+func updateNodeRetry(client corev1.NodeInterface, nodeName string, f func(*v1.Node)) (*v1.Node, error) {
+	var node *v1.Node
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		n, getErr := GetNode(client, node)
+		n, getErr := getNode(client, nodeName)
 		if getErr != nil {
 			return getErr
 		}
 
 		// Call the node modifier.
 		f(n)
 
-		_, err := client.Update(n)
+		var err error
+		node, err = client.Update(n)
 		return err
 	})
 	if err != nil {
 		// may be conflict if max retries were hit
-		return fmt.Errorf("Unable to update node %q: %v", node, err)
+		return nil, fmt.Errorf("Unable to update node %q: %v", node, err)
 	}
 
-	return nil
+	return node, nil
 }
 
-// setConfig sets the given annotation key, value pair.
-func setNodeAnnotations(client corev1.NodeInterface, node string, m map[string]string) error {
-	return updateNodeRetry(client, node, func(node *v1.Node) {
+func setNodeAnnotationsNodeBack(client corev1.NodeInterface, nodeName string, m map[string]string) (*v1.Node, error) {
```

Comment:
the `NodeBack` in the name is due to the func returning the node, right? Wondering whether we could name that differently...although that'd really just be personal preference, not really necessary. Maybe add a tiny comment?

Reply:
I suck at naming, a lot, so if you have any ideas I can change it lol

Reply:
revertNodeAnnotations? resetNodeAnnotations? fallBackNodeAnnotations? <-- i dont even know if those are better or worse tho.. :laughing:

Reply:
setNodeAnnotationsReturnNode?